### ISSUE TITLE
Umgang von Unterstrich in BibTeX

### ISSUE THREAD

**Issue by j3nsch** [OPUS4/application#497]
Der BibTeX-Parser kann beim Import nicht mit Unterstrichen umgehen. Diese müssen eigentlich Escaped werden,

**Comment by j3nsch** [comment]
Ich verwende BibTeX nicht, aber ich halte es für unrealistisch, dass Nutzer Unterstriche maskieren, wenn es die Programm nicht automatisch erledigen. Egal was der Standard sagt, kann ich gut verstehen, wenn Nutzer URLs im BibTeX haben wollen die ohne Bearbeitung direkt geöffnet werden können. Eine Maskierung von Unterstrichen in URLs  würde das verhindern.

Im Import Code sollten wir die LaTeX-Konvertierung nur für Felder verwenden, in denen LaTeX wahrscheinlich ist. Wenn das nicht reicht, oder zu häufig zu Problemen mit Sonderfällen führt, sollte wir die BibTeX-Dateien vorab scannen, und Unterstriche vorbehandeln.

**Comment by j3nsch** [comment]
Wir könnten Felder für LaTeX aktivieren bzw. deaktiveren, um nicht markierte URLs mit Unterstrichen ohne Fehler importieren zu können. Oder, falls das nicht reicht oder zu aufwendig ist, könnten wir versuchen das BibTeX scannen und eine Vorverarbeitung von URLs durchzuführen.

**Comment by j3nsch** [comment]
Beim BibTeX Export werde URLs auch nicht markiert. @alw-bsz Spricht etwas dagegen das hinzuzufügen, um möglichst Standard konformes BibTeX zu erzeugen oder würde das zu Problem an anderen Stellen führen?

**Comment by alw-bsz** [comment]
Nach unserem Kenntnisstand werden komplette URLs in BibTeX in der Regel nicht maskiert, sondern nur einzelne Bestandteile von Metadaten wie z.B. Sonderzeichen:
`[publisher = {Institut f{\"u}r Germanistik der Universit{\"a}t Szeged}]`

Die Form `\url{...}` ist vermutlich die reine LaTeX-Lehre.

**Comment by j3nsch** [comment]
Es ist also eher wahrscheinlich, dass andere Systeme damit nicht richtig umgehen können und wir deshalb darauf verzichten sollten?

Wie gesagt, es gibt andere Ansätze, um zu Exporten zu kommen, die wir auch wieder problemlos importieren können. Wir müssten die Felder mit URLs von der Interpretation als LaTeX ausschließen. Das würde ich nicht als Problem sehen.

**Comment by alw-bsz** [comment]
> Es ist also eher wahrscheinlich, dass andere Systeme damit nicht richtig umgehen können und wir deshalb darauf verzichten sollten?

Ja, genau das ist unsere Befürchtung und wir würden deshalb von der Umstellung abraten und stattdessen den Ansatz befürworten, URLs von der Interpretation als LaTeX auszunehmen.

**Comment by j3nsch** [comment]
Okay, danke. Ich denke der LaTeX-Schalter wird dann Teil des Mappings, so dass das jeder nach den eigenen Bedürfnissen anpassen kann. Im Mapping ist es zwar technisch etwas komplizierter, aber es macht dort einfach Sinn.
